### PR TITLE
Prefer builtin swap functions (updated alternative PR)

### PR DESCRIPTION
**Pull request by thbeu**
This is an updated PR for #95 to also remove the SwapWord function.

Let's see how the libshp behaves in GDAL test suite.

### CODE REVIEW

**Comment by thbeu** [comment]
Despite the CI fails (due to stricter warnings treated as error), it still is worth to check it in GDAL test suite.

File: shpopen.c [hunk 1]
```diff
@@ -100,65 +72,65 @@ void SHPAPI_CALL SHPWriteHeader(SHPHandle psSHP)
     uint32_t i32 = psSHP->nFileSize / 2; /* file size */
     ByteCopy(&i32, abyHeader + 24, 4);
 #if !defined(SHP_BIG_ENDIAN)
-    SwapWord(4, abyHeader + 24);
+    SHP_SWAP32(STATIC_CAST(uint32_t *, STATIC_CAST(void *, abyHeader + 24)));
```
Comment (rouault) [inline shpopen.c:75]: Sorry but all those casts look super ugly and make the code hardly readable

Comment (thbeu) [reply shpopen.c:75]: I know. First I need the GDAL test uiste result, then I will update,

Comment (rouault) [reply shpopen.c:75]: > I know. First I need the GDAL test suite result, then I will update,

well, you can easily test that yourself. clone OSGeo/GDAL, replace the shapefil.h, shpopen.c, dbfopen.c, shptree.c and sbnsearch.c files in ogr/ogrsf_frmts/shape and look at the CI results

**Comment by thbeu** [comment]
Closing in favour of #94.

**Pull request closed without merging**